### PR TITLE
Get outlook attachments command

### PR DESCRIPTION
**Pull request by openbots-su**

### CODE REVIEW

File: OpenBots.Commands/OpenBots.Commands.Microsoft/OpenBots.Commands.Outlook/GetOutlookEmailAttachmentsCommand.cs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+                    attachment.SaveAsFile(Path.Combine(attDirectory, attachment.FileName));
+                }
+                else if (!isEmbed)
+                {
+                    attachmentList.Add(attachment);
+                    attachment.SaveAsFile(Path.Combine(attDirectory, attachment.FileName));
+                }
+            }
+
+            attachmentList.StoreInUserVariable(engine, v_OutputUserVariableName);
```
Comment (openbots-ff) [inline OpenBots.Commands/OpenBots.Commands.Microsoft/OpenBots.Commands.Outlook/GetOutlookEmailAttachmentsCommand.cs:89]: The studio doesn't have any kind of support for Attachment types. You either need to add it to StringMethods.cs or just store a List<string> of file paths.

File: OpenBots.Commands/OpenBots.Commands.Microsoft/OpenBots.Commands.Outlook/GetOutlookEmailAttachmentsCommand.cs [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+            RenderedControls.AddRange(commandControls.CreateDefaultInputGroupFor("v_MailItem", this, editor));
+            RenderedControls.AddRange(commandControls.CreateDefaultInputGroupFor("v_AttachmentDirectory", this, editor));
+            RenderedControls.AddRange(commandControls.CreateDefaultDropdownGroupFor("v_IncludeEmbeddedImagesAsAttachments", this, editor));
+            RenderedControls.AddRange(commandControls.CreateDefaultOutputGroupFor("v_OutputUserVariableName", this, editor));
+
+            return RenderedControls;
+        }
+        public override string GetDisplayValue()
+        {
+            return base.GetDisplayValue() + $" [From '{v_MailItem}' - Store attachment list in '{v_OutputUserVariableName}']";
```
Comment (openbots-ff) [inline OpenBots.Commands/OpenBots.Commands.Microsoft/OpenBots.Commands.Outlook/GetOutlookEmailAttachmentsCommand.cs:105]: please follow capitalization conventions

File: OpenBots.Commands/OpenBots.Commands.Microsoft/OpenBots.Commands.Outlook/GetOutlookEmailAttachmentsCommand.cs (hunk 1 above)
Comment (openbots-ff) [inline OpenBots.Commands/OpenBots.Commands.Microsoft/OpenBots.Commands.Outlook/GetOutlookEmailAttachmentsCommand.cs:89]: The studio doesn't have any kind of support for Attachment types. You either need to add it to StringMethods.cs or just store a List of file paths.

File: OpenBots.Commands/OpenBots.Commands.Microsoft/OpenBots.Commands.Outlook/GetOutlookEmailAttachmentsCommand.cs (hunk 2 above)
Comment (openbots-ff) [inline OpenBots.Commands/OpenBots.Commands.Microsoft/OpenBots.Commands.Outlook/GetOutlookEmailAttachmentsCommand.cs:105]: please follow capitalization conventions

**Review by openbots-ff — CHANGES_REQUESTED** [review]
You'll need to create a similar command for smtp/imap emails

File: OpenBots.Commands/OpenBots.Commands.Microsoft/OpenBots.Commands.Outlook/GetOutlookEmailAttachmentsCommand.cs [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        [Remarks("")]
+        [Editor("ShowVariableHelper", typeof(UIAdditionalHelperType))]
+        [Editor("ShowFolderSelectionHelper", typeof(UIAdditionalHelperType))]
+        public string v_AttachmentDirectory { get; set; }
+
+        [Required]
+        [DisplayName("Include Embedded Images")]
+        [PropertyUISelectionOption("Yes")]
+        [PropertyUISelectionOption("No")]
+        [Description("Specify whether to consider images in body as attachments")]
```
Comment (openbots-ff) [inline OpenBots.Commands/OpenBots.Commands.Microsoft/OpenBots.Commands.Outlook/GetOutlookEmailAttachmentsCommand.cs:44]:
```suggestion
        [Description("Specify whether to consider images in body as attachments.")]
```

**Review by openbots-ff — CHANGES_REQUESTED** [review]

File: OpenBots.Commands/OpenBots.Commands.Microsoft/OpenBots.Commands.Outlook/GetOutlookEmailAttachmentsCommand.cs (hunk 3 above)
Comment (openbots-ff) [reply OpenBots.Commands/OpenBots.Commands.Microsoft/OpenBots.Commands.Outlook/GetOutlookEmailAttachmentsCommand.cs:44]: I can't remember if this property was missing punctuation in the get emails command (both for outlook and smtp), but if it was, it should be added there too.

**Review by openbots-ff — CHANGES_REQUESTED** [review]

File: OpenBots.Commands/OpenBots.Commands.Microsoft/OpenBots.Commands.Outlook/GetOutlookEmailAttachmentsCommand.cs (hunk 2 above)
Comment (openbots-su) [reply OpenBots.Commands/OpenBots.Commands.Microsoft/OpenBots.Commands.Outlook/GetOutlookEmailAttachmentsCommand.cs]: Addressed in 572971f

File: OpenBots.Commands/OpenBots.Commands.Microsoft/OpenBots.Commands.Outlook/GetOutlookEmailAttachmentsCommand.cs (hunk 1 above)
Comment (openbots-su) [reply OpenBots.Commands/OpenBots.Commands.Microsoft/OpenBots.Commands.Outlook/GetOutlookEmailAttachmentsCommand.cs:89]: Addressed in 572971f

File: OpenBots.Commands/OpenBots.Commands.SystemAutomation/OpenBots.Commands.Email/GetIMAPEmailAttachmentsCommand.cs [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+        [Remarks("")]
+        [Editor("ShowVariableHelper", typeof(UIAdditionalHelperType))]
+        [Editor("ShowFolderSelectionHelper", typeof(UIAdditionalHelperType))]
+        public string v_IMAPAttachmentDirectory { get; set; }
+
+        [Required]
+        [DisplayName("Include Embedded Images")]
+        [PropertyUISelectionOption("Yes")]
+        [PropertyUISelectionOption("No")]
+        [Description("Specify whether to consider images in body as attachments")]
```
Comment (openbots-ff) [inline OpenBots.Commands/OpenBots.Commands.SystemAutomation/OpenBots.Commands.Email/GetIMAPEmailAttachmentsCommand.cs:46]: Punctuation again. Did you check the other files like I asked before?

File: OpenBots.Commands/OpenBots.Commands.Microsoft/OpenBots.Commands.Microsoft.csproj [hunk 5]
```diff
@@ -87,6 +87,7 @@
     <Compile Include="OpenBots.Commands.Excel\ExcelWriteRowCommand.cs" />
     <Compile Include="OpenBots.Commands.Outlook\DeleteOutlookEmailCommand.cs" />
     <Compile Include="OpenBots.Commands.Outlook\ForwardOutlookEmailCommand.cs" />
+    <Compile Include="OpenBots.Commands.Outlook\GetOutlookEmailAttachmentsCommand.cs" />
```
Comment (openbots-ff) [inline OpenBots.Commands/OpenBots.Commands.Microsoft/OpenBots.Commands.Microsoft.csproj:90]: I don't see the IMAP command added to the SystemAutomation .csproj file.

File: OpenBots.Commands/OpenBots.Commands.Microsoft/OpenBots.Commands.Outlook/GetOutlookEmailAttachmentsCommand.cs [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+using OpenBots.Core.Properties;
+using OpenBots.Core.Utilities.CommonUtilities;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.ComponentModel.DataAnnotations;
+using System.Windows.Forms;
+using System.IO;
+
+namespace OpenBots.Commands.Microsoft.OpenBots.Commands.Outlook
```
Comment (openbots-ff) [inline OpenBots.Commands/OpenBots.Commands.Microsoft/OpenBots.Commands.Outlook/GetOutlookEmailAttachmentsCommand.cs:15]: please fix the namespaces on both commands

**Review by openbots-ff — CHANGES_REQUESTED** [review]

File: OpenBots.Commands/OpenBots.Commands.Microsoft/OpenBots.Commands.Microsoft.csproj (hunk 5 above)
Comment (openbots-su) [reply OpenBots.Commands/OpenBots.Commands.Microsoft/OpenBots.Commands.Microsoft.csproj:90]: Addressed in e312a98

File: OpenBots.Commands/OpenBots.Commands.Microsoft/OpenBots.Commands.Outlook/GetOutlookEmailAttachmentsCommand.cs (hunk 6 above)
Comment (openbots-su) [reply OpenBots.Commands/OpenBots.Commands.Microsoft/OpenBots.Commands.Outlook/GetOutlookEmailAttachmentsCommand.cs]: Addressed in e312a98

File: OpenBots.Commands/OpenBots.Commands.SystemAutomation/OpenBots.Commands.Email/GetIMAPEmailAttachmentsCommand.cs (hunk 4 above)
Comment (openbots-su) [reply OpenBots.Commands/OpenBots.Commands.SystemAutomation/OpenBots.Commands.Email/GetIMAPEmailAttachmentsCommand.cs]: All instances are updated in e312a98

**Review by openbots-ff — DISMISSED** [review]

**Review by openbots-ff — APPROVED** [review]

**Review by openbots-ch — APPROVED** [review]

**Pull request merged**